pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -321,6 +323,16 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 			ClusterName: name,
 			Endpoints:   localityLbEndpoints,
 		}
+	case cluster.Cluster_ORIGINAL_DST:
+		if features.PassthroughTargetPort {
+			if override, f := service.Attributes.PassthroughTargetPorts[uint32(port.Port)]; f {
+				c.LbConfig = &cluster.Cluster_OriginalDstLbConfig_{
+					OriginalDstLbConfig: &cluster.Cluster_OriginalDstLbConfig{
+						UpstreamPortOverride: wrappers.UInt32(override),
```

Comment:
how could this facilitate tls originate

Reply:
this helps rewrite traffic on port 80 to 443 so clients can just send normal requests instead of http on port 443. The PR description has an example of this and the PR has an e2e test doing this